File: src/main/java/ro/nicolaemariusghergu/queryit/model/Promotion.java
package ro.nicolaemariusghergu.queryit.model;

import lombok.*;
import org.hibernate.annotations.NotFound;
import org.hibernate.annotations.NotFoundAction;

import javax.persistence.*;
import java.io.Serializable;

@Getter
@Setter
@Entity
@NoArgsConstructor
@AllArgsConstructor
@ToString
@Table(name = "promotions")
public class Promotion implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @NotFound(action = NotFoundAction.IGNORE)
    @Column(name = "promotion_id", nullable = false)
    private Long id;

    @Column(name = "name")
    private String name;

    @Column(name = "description")
    private String description;

    @Column(name = "expire_date")
    private Long expireDate;

    @Column(name = "quantity_needed")
    private Integer quantityNeeded;

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Promotion promotion = (Promotion) o;

        return id.equals(promotion.id);
    }

    @Override
    public int hashCode() {
        return id.hashCode();
    }
}

File: src/main/kotlin/ro/nicolaemariusghergu/queryit/model/Promotion.kt
package ro.nicolaemariusghergu.queryit.model

import javax.persistence.*
import org.hibernate.annotations.NotFound
import org.hibernate.annotations.NotFoundAction

@Entity
@Table(name = "promotions")
class Promotion (
    @Column(name = "name")
    val name: String,

    @Column(name = "description")
    val description: String,

    @Column(name = "expire_date")
    val expireDate: Long,

    @Column(name = "quantity_needed")
    val quantityNeeded: Int,

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @NotFound(action = NotFoundAction.IGNORE)
    @Column(name = "promotion_id", nullable = false)
    val id: Long? = null
)